mlrun/projects/project.py

```diff
@@ -1520,6 +1533,10 @@ def register_artifacts(self):
                     is_retained_producer=is_retained_producer,
                 )
 
+    def update_artifact(self, artifact_object: Artifact):
+        artifacts_manager = self._get_artifact_manager()
+        artifacts_manager.update_artifact(artifact_object, artifact_object)
```

Comment:
1st parameter of `update_artifact` is the producer - I believe it should be `self` in this case.